external_search.py

```diff
@@ -17,12 +17,25 @@ class ExternalSearchIndex(object):
     work_document_type = 'work-type'
     __client = None
 
+    CURRENT_ALIAS_SUFFIX = '-current'
+    VERSION_RE = re.compile('-v([0-9]+)$')
+
+    @classmethod
+    def reset(cls):
+        """Resets the __client object to None so a new configuration
+        can be applied during object initialization.
+
+        This method is only intended for use in testing.
+        """
+        cls.__client = None
+
     def __init__(self, url=None, works_index=None):
     
         self.log = logging.getLogger("External search index")
 
         # By default, assume that there is no search index.
         self.works_index = None
+        from_configuration = False
```

Comment:
Yeah, you're right. I'm going to remove the `from_configuration` check here. It doesn't make a difference anyway, anymore.

Right now, there's no versioning implemented in this branch. The only time the version mapping would be accessed in the current arrangement is when the index doesn't exist.

If an `ELASTICSEARCH_INDEX_KEY` string is provided that doesn't exist as an index or an alias, there's no way for the initialization code to know which index to look for and upload documents to. This is the only case when it might create an index at a higher version than on that exists (but is unknown to the config).

Is there another case you see that I'm missing? Or could it be helpful for me to add a method to swap over the alias? That might make it clearer.